src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig

```diff
@@ -105,6 +106,61 @@
             });
         }
 
+        #createTableSearchFields() {
+            document.querySelectorAll('div.table-with-search-field').forEach((table, i) => {
+                const searchField = document.createElement('input');
+                searchField.type = 'search';
+                searchField.placeholder = 'search...';
+                searchField.id = `table-search-field-${i}`;
+                searchField.classList.add(`table-search-field-input`);
+                searchField.autocapitalize = 'off';
+                searchField.autocomplete = 'off';
+                searchField.autocorrect = 'off';
+                table.insertBefore(searchField, table.firstChild);
+
+                const labelField = document.createElement('label');
+                labelField.htmlFor = `table-search-field-${i}`;
+                labelField.classList.add(`table-search-field-label`);
+                labelField.innerHTML = 'Search inside the contents of the table';
+                table.insertBefore(labelField, table.firstChild);
+
+                searchField.addEventListener('input', () => {
+                    const query = searchField.value.toLowerCase();
+                    table.querySelectorAll('tbody tr').forEach((row) => {
+                        let rowMatchesQuery = false;
+                        for (const cell of row.querySelectorAll('td, th')) {
+                            const cellValue = cell.textContent || cell.innerText;
+                            if (cellValue.toLowerCase().includes(query)) {
+                                rowMatchesQuery = true;
+                                break;
+                            }
+                        }
+
+                        row.style.display = rowMatchesQuery ? '' : 'none';
+                    });
+
+                    /* if there are no results and all rows are hidden, show a message to avoid confusion */
+                    const allRows = table.querySelectorAll('tbody tr');
+                    const allRowsAreHidden = Array.from(allRows).every(row => 'none' === row.style.display);
+                    const noResultsElement = table.querySelector('.no-results-message');
+                    if (allRowsAreHidden) {
+                        if (null === noResultsElement) {
+                            const noResultsElement = document.createElement('p');
+                            noResultsElement.innerHTML = 'No results found.';
+                            noResultsElement.classList.add('no-results-message');
+                            table.appendChild(noResultsElement);
+                        } else {
+                            noResultsElement.style.display = '';
+                        }
+                    } else {
+                        if (null !== noResultsElement) {
+                            noResultsElement.style.display = 'none';
```

Comment:
Not sure i you want to, but this could be handled in CSS with `:has()` and `:not()` on the parent... if you use "`hidden`" instead of a style on the rows.

Reply:
Your proposal could work, but I prefer to create the "No results found" element dynamically because it will give us more flexibility in the future. Thanks.